### ISSUE TITLE
Create Schema view for endpoints

### ISSUE THREAD

**Issue by MissiaL** [tfranzel/drf-spectacular#19]
In [drf_yasg](https://github.com/axnsan12/drf-yasg#quickstart) I can get the schema and put it in endpoint automatically. Could you implement the same functionality?
```python
...
from rest_framework import permissions
from drf_yasg.views import get_schema_view
from drf_yasg import openapi

...

schema_view = get_schema_view(
   openapi.Info(
      title="Snippets API",
      default_version='v1',
      description="Test description",
      terms_of_service="https://www.google.com/policies/terms/",
      contact=openapi.Contact(email="contact@snippets.local"),
      license=openapi.License(name="BSD License"),
   ),
   public=True,
   permission_classes=(permissions.AllowAny,),
)

urlpatterns = [
   url(r'^swagger(?P<format>\.json|\.yaml)$', schema_view.without_ui(cache_timeout=0), name='schema-json'),
   url(r'^swagger/$', schema_view.with_ui('swagger', cache_timeout=0), name='schema-swagger-ui'),
   url(r'^redoc/$', schema_view.with_ui('redoc', cache_timeout=0), name='schema-redoc'),
   ...
]
```
Now I use the template and do it manually. I read the schema from the file and then use the swagger template

**Comment by tfranzel** [comment]
don't we already have that with https://github.com/tfranzel/drf-spectacular/blob/master/drf_spectacular/views.py#L16 ?

Only difference is you have to setup the settings in SPECTACULAR_SETTINGS and just load SpectacularApiView in your urlpatterns.

the caching mechanism that yasg provides is missing though.

**Comment by MissiaL** [comment]
Wow! 
I did not notice this! Thanks!

**Comment by aniforprez** [comment]
I suggest this be documented in the README or somewhere. There's also no documentation for the settings available in the package

**Comment by tfranzel** [comment]
@aniforprez actually there is "documentation" for the settings: https://drf-spectacular.readthedocs.io/en/latest/settings.html

you made a good point though. the readme could be better and some feature are not obvious from the get-go. i updated https://github.com/tfranzel/drf-spectacular#take-it-for-a-spin (schema view) and usage (for the settings).

**Comment by aniforprez** [comment]
Very nice! I also recommend at some point adding some examples with swagger or redoc or both so there's immediately something they can visually look at rather than only getting the file which is hard to parse through visually

Btw nice work!

**Comment by vladdoster** [comment]
How exactly do I get it to serve the UI? I added it to my URLS but everytimg I go to the path it just downloads the schema?

**Comment by tfranzel** [comment]
@vladdoster we do not serve the UI ourselves:

swagger-ui: 
$ ./manage.py spectacular --file schema.yml
$ docker run -p 80:8080 -e SWAGGER_JSON=/schema.yml -v ${PWD}/schema.yml:/schema.yml swaggerapi/swagger-ui

redoc:
serve this html with your api https://github.com/Redocly/redoc#tldr and replace the schema url in it.

**Comment by MissiaL** [comment]
I had to take the implementation from the `drf_yasg` package
It’s actually very convenient when you can display UI  on your  django server

I added a [swagger template](https://www.django-rest-framework.org/topics/documenting-your-api/#a-minimal-example-with-swagger-ui) and give it a schema

**Comment by tfranzel** [comment]
i was not aware that serving swagger-ui through a template like redoc is that simple. in that case it might be not a bad idea to add that convenience. i'll have a look

**Comment by vladdoster** [comment]
Thank you so much @MissiaL && @tfranzel !

**Comment by tfranzel** [comment]
added swagger&redoc and updated the README with new instructions. let me know if that works for you.

**Comment by vladdoster** [comment]
@tfranzel will give it a go tonight

**Comment by tfranzel** [comment]
i used both redoc and swagger today. looks good to me.
